controllers/drplacementcontrol_controller.go

```diff
@@ -953,56 +1018,146 @@ func (d *DRPCInstance) switchToPreferredCluster(drState rmn.DRState) (bool, erro
 	d.setDRState(drState)
 	setMetricsTimerFromDRState(drState, timerStart)
 
-	// During failback or relocation, both clusters should be up and both must be secondaries before we proceed
-	ensured, err := d.processVRGForSecondaries()
-	if err != nil || !ensured {
-		return !done, err
+	// Skip switching to secondary if the preferredCluster already is or in progress to becoming primary
+	if !d.isVRGReadyForPlacementTask(preferredCluster) {
```

Comment:
If there is no VRG on the `preferredCluster` this will return `true` and skip moving other clusters to Secondary before placing the workload, which would cause relocate issues, as we need the workload to stop and move to Secondary first.

Reply:
That's true.  I noticed that.